resources/js/app.js

```diff
@@ -124,4 +124,262 @@ window.chart = (async function () {
+            const title = document.createElement("span");
+            title.innerText = chapter.title;
+            leftDiv.appendChild(playIcon);
+            leftDiv.appendChild(title);
+            const playButton = document.createElement("button");
+            playButton.setAttribute("data-url", chapter.video_url);
+            playButton.className = "btn previewLink";
+            playButton.innerText = "play";
+
+            playButton.addEventListener("click", handlePreviewLink);
+            const videoDuration = document.createElement("span");
+            videoDuration.innerText =
+                secondsToHMS(chapter.video_duration) || "-";
+            rightDiv.appendChild(playButton);
+            rightDiv.appendChild(videoDuration);
+
+            container.appendChild(leftDiv);
+            container.appendChild(rightDiv);
+
+            fragment.appendChild(container);
+        });
+    }
+
+    return fragment;
+}
+
+function handlePreviewLink(e) {
+    const { url } = e.target.dataset;
+    const previewLinks = document.querySelectorAll(".previewLink");
+    previewLinks.forEach((link) => {
+        const grandparent = link.parentElement.parentElement;
+        grandparent.classList.remove("text-primary");
+    });
+    e.target.parentElement.parentElement.classList.add("text-primary");
+    renderVideo(url);
+}
+
+function secondsToHMS(seconds) {
+    if (!seconds) return null;
+    const hours = Math.floor(seconds / 3600);
+    const minutes = Math.floor((seconds % 3600) / 60);
+    const remainingSeconds = seconds % 60;
+
+    if (hours > 0) {
+        const formattedHours = hours.toString().padStart(2, "0");
+        const formattedMinutes = minutes.toString().padStart(2, "0");
+        const formattedSeconds = remainingSeconds.toString().padStart(2, "0");
+        return `${formattedHours}:${formattedMinutes}:${formattedSeconds}`;
+    } else {
+        const formattedMinutes = minutes.toString().padStart(2, "0");
+        const formattedSeconds = remainingSeconds.toString().padStart(2, "0");
+        return `${formattedMinutes}:${formattedSeconds}`;
+    }
+}
+
+window.chapterVideoFileName = "";
+window.chapterVideoURL = "";
+window.chapterVideoFile = null;
+window.youtubeURL = "";
+window.isSubmitVideoButtonDisabled = true;
+window.chapterVideoDuration = 0;
+
+window.handleVideoUrlChange = async function (e) {
+    const {
+        data: { duration },
+    } = await getVideoInfo(e.target.value);
+    this.chapterVideoDuration = duration;
+    this.isSubmitVideoButtonDisabled = false;
+};
+
+async function getVideoInfo(videoUrl) {
+    try {
+        const response = await axios.post(
+            "https://echo-tube.vercel.app/get-video-info",
+            {
+                videoUrl: videoUrl,
+            },
+        );
+
+        return response.data;
+    } catch (error) {
+        console.error("Error fetching video info:", error);
+        return null;
+    }
+}
```

Comment:
Run this on the backend side instead, because we might limit the API using apikey someday.